### PR TITLE
Coverage for core parser

### PR DESCRIPTION
**Pull request by abdrabrcom**
as per issue #110 

increased total unit testing code coverage to 69.3%

### CODE REVIEW

File: com.ca.lsp.cobol/lsp-core-cobol-parser/src/test/java/com/ca/lsp/core/cobol/preprocessor/sub/line/rewriter/impl/CobolLineIndicatorProcessorImplTest.java [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+  private CobolLine TRAILING_QUOTE_CONTINUATION_LINE = new CobolLine();
+  private CobolLine EMPTY_CONTINUATION_LINE_TO_TEST = new CobolLine();
+  private CobolLine SINGLE_CONTINUATION_LINE_TO_TEST = new CobolLine();
+  private CobolLine GOOD_CONTINUATION_LINE_TO_TEST = new CobolLine();
+  private CobolLine BAD_CONTINUATION_LINE_TO_TEST = new CobolLine();
+  private CobolLine COMMENT_LINE_TO_TEST = new CobolLine();
+  private CobolLine CONTINUATION_LINE_WITH_END_QUOTES = new CobolLine();
+  private CobolLine COMPILER_DIRECTIVE_LINE_TO_TEST = new CobolLine();
+  private CobolLine NORMAL_LINE_TO_TEST = new CobolLine();
+  private CobolLine DEBUG_LINE_TO_TEST = new CobolLine();
```
Comment (temanbrcom) [inline com.ca.lsp.cobol/lsp-core-cobol-parser/src/test/java/com/ca/lsp/core/cobol/preprocessor/sub/line/rewriter/impl/CobolLineIndicatorProcessorImplTest.java]: These fields may be converted to local variables

File: com.ca.lsp.cobol/lsp-core-cobol-parser/src/test/java/com/ca/lsp/core/cobol/preprocessor/sub/line/rewriter/impl/CobolLineIndicatorProcessorImplTest.java [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+            EMPTY_CONTINUATION_LINE_TO_TEST,
+            COMMENT_LINE_TO_TEST,
+            COMPILER_DIRECTIVE_LINE_TO_TEST,
+            NORMAL_LINE_TO_TEST,
+            DEBUG_LINE_TO_TEST,
+            CONTINUATION_LINE_WITH_END_QUOTES);
+    CobolLineIndicatorProcessorImpl processor = new CobolLineIndicatorProcessorImpl();
+    processor.processLines(listOfLines);
+
+    // assertEquals("RANDOM TEXT",processor.processLine(CONTINUATION_LINE));
```
Comment (temanbrcom) [inline com.ca.lsp.cobol/lsp-core-cobol-parser/src/test/java/com/ca/lsp/core/cobol/preprocessor/sub/line/rewriter/impl/CobolLineIndicatorProcessorImplTest.java]: Please, remove the commented line

File: com.ca.lsp.cobol/lsp-core-cobol-parser/src/test/java/com/ca/lsp/core/cobol/preprocessor/sub/line/rewriter/impl/CobolLineIndicatorProcessorImplTest.java [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+import static com.ca.lsp.core.cobol.preprocessor.sub.CobolLineTypeEnum.DEBUG;
+import static com.ca.lsp.core.cobol.preprocessor.ProcessingConstants.WS;
+import static org.junit.Assert.assertEquals;
+
+import com.ca.lsp.core.cobol.preprocessor.sub.CobolLine;
+
+import java.util.Arrays;
+import java.util.List;
+import org.junit.Test;
+
```
Comment (temanbrcom) [inline com.ca.lsp.cobol/lsp-core-cobol-parser/src/test/java/com/ca/lsp/core/cobol/preprocessor/sub/line/rewriter/impl/CobolLineIndicatorProcessorImplTest.java]: Please, provide the javadoc description for this and following classes

File: com.ca.lsp.cobol/lsp-core-cobol-parser/src/test/java/com/ca/lsp/core/cobol/preprocessor/sub/util/PreprocessorStringUtilsTest.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+public class PreprocessorStringUtilsTest {
+
+  final String TEST_SINGLE_QUOTE_STRING = "\'TESTING IF ALL QUOTES WILL BE REMOVED\'";
+  final String TEST_DOUBLE_QUOTE_STRING = "\"TESTING IF ALL QUOTES WILL BE REMOVED\"";
+  final String EXPECTED_STRING = "TESTING IF ALL QUOTES WILL BE REMOVED";
+
+  @Test
+  public void trimQuotesTest() {
+    assertEquals(EXPECTED_STRING,PreprocessorStringUtils.trimQuotes(TEST_SINGLE_QUOTE_STRING));
+    assertEquals(EXPECTED_STRING,PreprocessorStringUtils.trimQuotes(TEST_DOUBLE_QUOTE_STRING));
```
Comment (temanbrcom) [inline com.ca.lsp.cobol/lsp-core-cobol-parser/src/test/java/com/ca/lsp/core/cobol/preprocessor/sub/util/PreprocessorStringUtilsTest.java]: These lines are not formatted properly

File: com.ca.lsp.cobol/lsp-core-cobol-parser/src/test/java/com/ca/lsp/core/cobol/strategy/CobolErrorStrategyTest.java [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+import org.antlr.v4.runtime.RecognitionException;
+import org.antlr.v4.runtime.Token;
+import org.antlr.v4.runtime.TokenStream;
+import org.antlr.v4.runtime.VocabularyImpl;
+import org.antlr.v4.runtime.misc.IntervalSet;
+import org.junit.Test;
+
+public class CobolErrorStrategyTest {
+  @Test
+  public void NoViableAltExceptionTest() {
```
Comment (temanbrcom) [inline com.ca.lsp.cobol/lsp-core-cobol-parser/src/test/java/com/ca/lsp/core/cobol/strategy/CobolErrorStrategyTest.java]: Method names should be in a camel-case starting with a lower-case character.
The same for the other methods here.

File: com.ca.lsp.cobol/lsp-core-cobol-parser/src/test/java/com/ca/lsp/core/cobol/preprocessor/sub/line/rewriter/impl/CobolLineIndicatorProcessorImplTest.java (hunk 1 above)
Comment (abdrabrcom) [reply com.ca.lsp.cobol/lsp-core-cobol-parser/src/test/java/com/ca/lsp/core/cobol/preprocessor/sub/line/rewriter/impl/CobolLineIndicatorProcessorImplTest.java]: Done

File: com.ca.lsp.cobol/lsp-core-cobol-parser/src/test/java/com/ca/lsp/core/cobol/strategy/CobolErrorStrategyTest.java (hunk 5 above)
Comment (abdrabrcom) [reply com.ca.lsp.cobol/lsp-core-cobol-parser/src/test/java/com/ca/lsp/core/cobol/strategy/CobolErrorStrategyTest.java]: Done

File: com.ca.lsp.cobol/lsp-core-cobol-parser/src/test/java/com/ca/lsp/core/cobol/preprocessor/sub/util/PreprocessorStringUtilsTest.java (hunk 4 above)
Comment (abdrabrcom) [reply com.ca.lsp.cobol/lsp-core-cobol-parser/src/test/java/com/ca/lsp/core/cobol/preprocessor/sub/util/PreprocessorStringUtilsTest.java]: Done

File: com.ca.lsp.cobol/lsp-core-cobol-parser/src/test/java/com/ca/lsp/core/cobol/preprocessor/sub/line/rewriter/impl/CobolLineIndicatorProcessorImplTest.java (hunk 2 above)
Comment (abdrabrcom) [reply com.ca.lsp.cobol/lsp-core-cobol-parser/src/test/java/com/ca/lsp/core/cobol/preprocessor/sub/line/rewriter/impl/CobolLineIndicatorProcessorImplTest.java]: Done

File: com.ca.lsp.cobol/lsp-core-cobol-parser/src/test/java/com/ca/lsp/core/cobol/preprocessor/sub/line/rewriter/impl/CobolLineIndicatorProcessorImplTest.java (hunk 3 above)
Comment (abdrabrcom) [reply com.ca.lsp.cobol/lsp-core-cobol-parser/src/test/java/com/ca/lsp/core/cobol/preprocessor/sub/line/rewriter/impl/CobolLineIndicatorProcessorImplTest.java]: Done

**Comment by zacanbrcom** [comment]
I think we should split the CobolLineIndicatorProcessorImplTest asserts in single unit tests and make CobolLineIndicatorProcessorImplTest a kind of suite for indicator preprocessor. In that way also the many instances of CobolLine could be drastically reduced.

File: com.ca.lsp.cobol/lsp-core-cobol-parser/src/test/java/com/ca/lsp/core/cobol/preprocessor/sub/line/rewriter/impl/CobolLineIndicatorProcessorImplTest.java [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+    goodContinuationLineToTest.setSuccessor(continuationLine);
+    goodContinuationLineToTest.setIndicatorArea(WS);
+    goodContinuationLineToTest.setContentAreaA("       \"RANDOM TEXT   ");
+    goodContinuationLineToTest.setContentAreaB("        ");
+
+    singleContinuationLineToTest.setType(CONTINUATION);
+    singleContinuationLineToTest.setSuccessor(continuationLine);
+    singleContinuationLineToTest.setIndicatorArea(WS);
+    singleContinuationLineToTest.setContentAreaA("       'RANDOM TEXT   ");
+    singleContinuationLineToTest.setContentAreaB("        ");
```
Comment (zacanbrcom) [inline com.ca.lsp.cobol/lsp-core-cobol-parser/src/test/java/com/ca/lsp/core/cobol/preprocessor/sub/line/rewriter/impl/CobolLineIndicatorProcessorImplTest.java]: Do we really need to emphasize this "to test"?

File: com.ca.lsp.cobol/lsp-core-cobol-parser/src/test/java/com/ca/lsp/core/cobol/preprocessor/sub/line/rewriter/impl/CobolLineIndicatorProcessorImplTest.java [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+        processor.processLines(listOfLines).get(8).getIndicatorArea()
+            + processor.processLines(listOfLines).get(8).getContentArea());
+    assertEquals(
+        WS + "     DEBUG LINE HERE",
+        processor.processLines(listOfLines).get(9).getIndicatorArea()
+            + processor.processLines(listOfLines).get(9).getContentArea());
+    assertEquals(
+        WS + "RANDOM TEXT \"",
+        processor.processLines(listOfLines).get(10).getIndicatorArea()
+            + processor.processLines(listOfLines).get(10).getContentArea());
```
Comment (zacanbrcom) [inline com.ca.lsp.cobol/lsp-core-cobol-parser/src/test/java/com/ca/lsp/core/cobol/preprocessor/sub/line/rewriter/impl/CobolLineIndicatorProcessorImplTest.java]: We should add a line of comment (or a javadoc if defined in single unit test) to explain in detail what we are trying to verify.

File: com.ca.lsp.cobol/lsp-core-cobol-parser/src/test/java/com/ca/lsp/core/cobol/preprocessor/sub/util/PreprocessorStringUtilsTest.java [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+package com.ca.lsp.core.cobol.preprocessor.sub.util;
+
+import static org.junit.Assert.assertEquals;
+import org.junit.Test;
+
+public class PreprocessorStringUtilsTest {
+
+  static final String TEST_SINGLE_QUOTE_STRING = "\'TESTING IF ALL QUOTES WILL BE REMOVED\'";
+  static final String TEST_DOUBLE_QUOTE_STRING = "\"TESTING IF ALL QUOTES WILL BE REMOVED\"";
+  static final String EXPECTED_STRING = "TESTING IF ALL QUOTES WILL BE REMOVED";
```
Comment (temanbrcom) [inline com.ca.lsp.cobol/lsp-core-cobol-parser/src/test/java/com/ca/lsp/core/cobol/preprocessor/sub/util/PreprocessorStringUtilsTest.java]: These constants should be private

File: com.ca.lsp.cobol/lsp-core-cobol-parser/src/test/java/com/ca/lsp/core/cobol/preprocessor/sub/line/rewriter/impl/CobolLineIndicatorProcessorImplTest.java (hunk 3 above)
Comment (temanbrcom) [reply com.ca.lsp.cobol/lsp-core-cobol-parser/src/test/java/com/ca/lsp/core/cobol/preprocessor/sub/line/rewriter/impl/CobolLineIndicatorProcessorImplTest.java]: Every class should have a javadoc comment *before* the class name describing its purpose

File: com.ca.lsp.cobol/lsp-core-cobol-parser/src/test/java/com/ca/lsp/core/cobol/preprocessor/sub/line/rewriter/impl/CobolLineIndicatorProcessorImplTest.java (hunk 6 above)
Comment (abdrabrcom) [reply com.ca.lsp.cobol/lsp-core-cobol-parser/src/test/java/com/ca/lsp/core/cobol/preprocessor/sub/line/rewriter/impl/CobolLineIndicatorProcessorImplTest.java]: Removed

File: com.ca.lsp.cobol/lsp-core-cobol-parser/src/test/java/com/ca/lsp/core/cobol/preprocessor/sub/util/PreprocessorStringUtilsTest.java (hunk 8 above)
Comment (abdrabrcom) [reply com.ca.lsp.cobol/lsp-core-cobol-parser/src/test/java/com/ca/lsp/core/cobol/preprocessor/sub/util/PreprocessorStringUtilsTest.java]: Corrected

File: com.ca.lsp.cobol/lsp-core-cobol-parser/src/test/java/com/ca/lsp/core/cobol/preprocessor/sub/line/rewriter/impl/CobolLineIndicatorProcessorImplTest.java (hunk 7 above)
Comment (abdrabrcom) [reply com.ca.lsp.cobol/lsp-core-cobol-parser/src/test/java/com/ca/lsp/core/cobol/preprocessor/sub/line/rewriter/impl/CobolLineIndicatorProcessorImplTest.java]: reworked the junit test and added proper javadoc information for each test class

**Pull request merged**